### ISSUE TITLE
Improve automatic differentiation documentation

### ISSUE THREAD

**Issue by Rombur** [dealii/dealii#6172]
@jppelteret Would it be possible to improve the documentation [here](http://dealii.org/developer/doxygen/deal.II/group__auto__symb__diff.html). There are a couple of `Todo` left.

**Comment by davydden** [comment]
blocks #6156

**Comment by bangerth** [comment]
@jppelteret -- are you still on this?

**Comment by jppelteret** [comment]
@bangerth Yes, but as I had explained I'm struggling a bit for time (and am OOO at the moment). I should be able to tackle this on Saturday.
